/*
 * Copyright 2017-2024 O2 Czech Republic, a.s.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
syntax = "proto3";

package cz.o2.proxima.example.user;

option java_outer_classname = "User";

/* details of user */
message Details {

  /* Name of user */
  string name = 1;

  /* User's username */
  string userName = 2;

  /* User's email */
  string email = 3;

}

/*
 * Preferences of user.
 * This is heavily dependent on the actual preference extraction algorithm used
 * and is not covered by this example, so let's just suppose, that
 * user's preferences can be expressed as a vector in some
 * latent vector space.
 */
message Preferences {

  /* Identifier of algorithm used for the latent vector extraction. */
  string algorithm = 1;

  /* The latent vector. */
  repeated float vector = 2;

}

